Nuevo intento

SanaBrian opened the pull request.

SanaBrian committed:
Nuevo intento

SanaBrian commented:
Hola Ariel, perdón por la demora, creía que lo había subido bien. Espero que ahora sí puedan agregar comentarios

alutenberg commented:
> Hola Ariel, perdón por la demora, creía que lo había subido bien. Espero que ahora sí puedan agregar comentarios

Sí, sí, lo subiste bien! Gracias!

Ya está incluido aca en celeste:

https://docs.google.com/spreadsheets/d/1cOt5CU1fZoQtIt7TfgWl9n3rmaLqNTvy04iO_wj7IZ0/edit?gid=1187394869#gid=1187394869

EFilomena commented on README.md:
```diff
@@ -1 +1,32 @@
+
+El sistema debe controlar la temperatura dentro de una cámara térmica y mantenerla dentro de una
+temperatura máxima y mínima preestablecida.
+Cada 20 minutos deberá enviar un registro de temperatura a la PC.
```
No sería conveniente enviar los datos cada un intervalo menor? Un minuto o algo así.
La PC, puede hacer con los datos lo que quiera... incluso descartarlos si no los necesita

EFilomena commented on main.cpp:
```diff
@@ -0,0 +1,131 @@
+//-------------- Declaración de variables ---------------
+
+float maxTempValue = 50.0;
+float minTempValue = 25.0;
```
Estos valores son fijos o se pueden modificar?

SanaBrian replied:
Estos valores se pueden cambiar. La idea  es poder cambiarlos con el teclado matricial en el próximo TP

EFilomena replied:
Genial, muy buena idea

EFilomena commented on main.cpp:
```diff
@@ -0,0 +1,131 @@
+}
+
+void initOuputs(){
+    maxTempLED = OFF;
```
ENCENDIDO o PRENDIDO --- APAGADO quedaría más lindo no??

SanaBrian replied:
Se entendería más lo que está haciendo. Lo voy a cambiar.

EFilomena commented on main.cpp:
```diff
@@ -0,0 +1,131 @@
+
+int tempConverter(float v){                                    //Convierte señal eléctrica a ºC y actualiza el valor de temperatura
+    float tempC = v * 3.3/0.01;
+    return tempC;
```
tempC es float y la función devuelve un entero. Ojo con eso!

EFilomena commented on main.cpp:
```diff
@@ -0,0 +1,131 @@
+    lm35ReadingsSum = 0.0;
+    for (i = 0; i < NUMBER_OF_AVG_SAMPLES; i++) {
+        lm35ReadingsSum = lm35ReadingsSum + lm35ReadingsArray[i];
+    }
```
Fijate que los primeros promedios (NUMBER_OF_AVG_SAMPLES-1) vas a promediar el valor leído con algunos ceros.
Esto hace que el promedio sea erroneo.
Hay varias soluciones a ésto. Una puede ser inicializar el vector con la primera lectura, otra usar una bandera que indique cuándo el promedio es válido, otra es promediar sólo los valores que tenés (hasta completar la carga completa del vector)

SanaBrian replied:
Voy por la opción de promediar con la primera lectura.

EFilomena replied:
Perfecto!, probá como funciona y comentame

EFilomena commented on main.cpp:
```diff
@@ -0,0 +1,131 @@
+        while (lm35TempC <= avgTempValue){
+            minTempLED = ON;
+            samplerAverager();   
+        }
```
Esta función es bloqueante también!

EFilomena commented on main.cpp:
```diff
@@ -0,0 +1,131 @@
+        uartUsb.write("Alcanzó la máxima temperatura\r\n", 30); 
+        while (lm35TempC > avgTempValue){
+            maxTempLED = ON;
+            samplerAverager();
```
Esta función es bloqueante. No sale de este while() mientras que la temp sea mayor al umbral fijado.
Deberías pensar en modificar esto. Una idea sería armar una máquina de estados con una variable que indique en qué estado está tu sistema: (ENFRIANDO, CALENTANDO, NORMAL) y en base al estado, monitorear la temperatura y tomar las decisiones. Existen otras soluciones, dale con la que te sientas mas cómodo.......

SanaBrian replied:
Si, acá tuve un tema de cómo resolverlo aplicando if() para que no sea bloqueante pero opté ir por acá y darle importancia al enfriado o calentado y no atender otras operaciones. Voy a intentar trabajar con la máquina de estado para resolverlo.

EFilomena replied:
Ok

EFilomena commented on main.cpp:
```diff
@@ -0,0 +1,131 @@
+
+
+void checkOverHeating(){                                        //Chequea si la temperatura está por arriba del limite
+    if(lm35TempC >= maxTempValue){
```
Ojo con la comparación de float.  lm35TempC  nunca va a ser igual  a maxTempValue . De todas maneras la comparación si funciona

SanaBrian replied:
A corregir ese detalle

SanaBrian commented:
Les consulto lo siguiente:

- Debo hacer un "merge pull resquest"?
- Ya puedo cerrar este pull resquest?
- Luego de hacer las modificaciones, realizo un nuevo pedido de revisión o sigo dejando este pedido abierto?

Gracias por los comentarios

scaprile commented:
> Les consulto lo siguiente:
> 
>     * Debo hacer un "merge pull resquest"?
> 
>     * Ya puedo cerrar este pull resquest?
> 
>     * Luego de hacer las modificaciones, realizo un nuevo pedido de revisión o sigo dejando este pedido abierto?
> 
> 
> Gracias por los comentarios

Depende.
Si se te pidieron cambios, tenés que hacer cambios. Como dicen los mails y el video que mandó Ariel, los cambios se hacen en el mismo branch del PR. Si lo cerrás, no podés, o sea que ahí tenés tu respuesta.
Si se te dejaron comentarios, el revisor no aprobó todavía, cuando contestes los comentarios, verá si te pide cambios o te aprueba. Si lo cerrás, no podés, ni puede, o sea que ahí tenés tu respuesta.
Si los 3 revisores te aprobaron, no tenés que hacer nada más sobre ese branch. El código está aprobado y como dicen los mails y el video que mandó Ariel, pasa al tronco principal. Eso se hace con un merge, o sea que ahí tenés tu respuesta.

:smiley:

SanaBrian committed:
Correcciones sugeridas y uso de máquina de estados

EFilomena reviewed (commented):
Se ve mucho mejor Brian, mucho más prolijo y no encuentro nada que objetar. 
No encuentro la implementaricón de stateUpdate() puede ser????

SanaBrian commented:
Buen día Eduardo. 
La función `stateUpdate()` está implementada dentro del _main_, dentro del bucle _while_, debajo de la función `checkOverCooling()`. Decime si pudiste visualizarla porque sino subí mal el archivo.

SanaBrian committed:
Update main.cpp

Borré el signo = porque ingresaría la lectura en una posición inexistente en el array

EFilomena commented:
En ese lugar está la llamada a la función, no veo la implementación de la misma

SanaBrian committed:
Update main.cpp

Se agrega función faltante

SanaBrian commented:
Es verdad, se me pasó agregarla. Ya la agregué.

EFilomena reviewed (approved):
Genial
